### PR TITLE
build(deps): bump react-native from 0.70.14 to 0.72.10

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-574] 

### Description

Used [react native upgade helper](https://react-native-community.github.io/upgrade-helper/?from=0.70.14&to=0.72.10) for this and most of the diff comes from this.

- Bumps react native to `0.72.10`
- adds `@types/react` earlier version to match eigen and fixes some type issues after that
- adds some deps but its the suggestion from the upgrade helper 

Both apps work ~ need to do a qa for sure before merging will trigger a canary version to test it out in eigen and folio

<details>
  <summary>📦 Published PR as canary version: <code>13.0.30--canary.175.1112.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @artsy/palette-mobile@13.0.30--canary.175.1112.0
  # or 
  yarn add @artsy/palette-mobile@13.0.30--canary.175.1112.0
  ```
</details>

[PHIRE-574]: https://artsyproduct.atlassian.net/browse/PHIRE-574?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -38,3 +38,8 @@ reactNativeArchitectures=armeabi-v7a,arm64-v8a,x86,x86_64
 # to write custom TurboModules/Fabric components OR use libraries that
 # are providing them.
 newArchEnabled=false
+
+
+# Use this property to enable or disable the Hermes JS engine.
+# If set to false, you will be using JSC instead.
+hermesEnabled=false
```
[the remark this answers is not in the document]
Comment (gkartalis) [reply android/gradle.properties:45]: good catch, we want that true

File: android/app/src/release/java/com/palettemobile/ReactNativeFlipper.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * <p>This source code is licensed under the MIT license found in the LICENSE file in the root
+ * directory of this source tree.
+ */
+package com.rndiffapp;
```
Comment (brainbicycle) [inline android/app/src/release/java/com/palettemobile/ReactNativeFlipper.java:7]: think the package should be the package name for the palette app

Comment (gkartalis) [reply android/app/src/release/java/com/palettemobile/ReactNativeFlipper.java:7]:
```suggestion
package com.palettemobile;
```

File: android/gradle.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (gkartalis) [reply android/gradle.properties:45]:
```suggestion
hermesEnabled=true
```

File: android/app/src/release/java/com/palettemobile/ReactNativeFlipper.java (hunk 2 above)
Comment (brainbicycle) [inline android/app/src/release/java/com/palettemobile/ReactNativeFlipper.java]:
```suggestion
package com.palettemobile;
```

**Comment by gkartalis** [comment]
🥷

**Review by MounirDhahri — APPROVED** [review]
LGTM! thanks for making this update

**Comment by gkartalis** [comment]
once [this PR on artsy/orbs](https://github.com/artsy/orbs/pull/163) is merged and published we will want to change the auto orb to point to 2.2.0: auto: instad of artsy/auto@dev:eabf0bd05f85d0155319684487dd4fa9 currently pointing at canary

**Comment by ovasdi** [comment]
> once [this PR on artsy/orbs](https://github.com/artsy/orbs/pull/163) is merged and published we will want to change the auto orb to point to 2.2.0: auto: instad of `artsy/auto@dev:eabf0bd05f85d0155319684487dd4fa9` currently pointing at canary

@gkartalis [published](https://app.circleci.com/pipelines/github/artsy/orbs/584/workflows/9d6eac7c-bcdb-46a5-8d15-b859382fab74/jobs/945?invite=true#step-103-174_37)

**Comment by artsyit** [comment]
:rocket: PR was released in [`v13.0.30`](https://github.com/artsy/palette-mobile/releases/tag/v13.0.30) :rocket:

**Pull request merged**